sdks/python/apache_beam/testing/util.py

```diff
@@ -261,6 +261,19 @@ def assert_that(
   """
   assert isinstance(actual, pvalue.PCollection), (
       '%s is not a supported type for Beam assert' % type(actual))
+  pipeline = actual.pipeline
+  if getattr(pipeline, 'result', None) is not None:
+    # The pipeline was already run. The user most likely called assert_that
+    # after the pipeleline context.
+    raise RuntimeError(
+        'assert_that must be used within a beam.Pipeline context')
+
+  # If label is already in use, just append a number to it.
```

Comment:
Can you add a comment as to why this is OK to do in this context?

Reply:
I put:
```
  # Usually, the uniqueness of the label is left to the pipeline
  # writer to guarantee. Since we're in a testing context, we'll
  # just automatically append a number to the label if it's
  # already in use.
```
Let me know if something else would be better

Reply:
Maybe add something about not needing stability of names across pipeline update in a testing context?